packages/yarnpkg-builder/sources/commands/build/bundle.ts

```diff
@@ -88,96 +87,29 @@ export default class BuildBundleCommand extends Command {
       forgettableNames: new Set([MessageName.UNNAMED]),
     }, async report => {
       await report.startTimerPromise(`Building the CLI`, async () => {
-        const progress = StreamReport.progressViaCounter(1);
-        report.reportProgress(progress);
-
-        const prettyWebpack = structUtils.prettyIdent(configuration, structUtils.makeIdent(null, `webpack`));
-
-        const compiler = webpack(makeConfig({
-          context: basedir,
-          entry: `./sources/cli.ts`,
-
-          bail: true,
-
-          ...!this.noMinify && {
-            mode: `production`,
+        const valLoad = (p: string, values: any) => {
+          const fn = require(p.replace(/.ts$/, `.val.js`));
+          return fn(values).code;
+        };
+
+        const valLoader: Plugin = {
+          name: `val-loader`,
+          setup(build) {
+            build.onLoad({filter: /\/getPluginConfiguration\.ts$/}, async args => ({
+              contents: valLoad(args.path, {modules, plugins}),
+              loader: `default`,
+            }));
           },
-
-          ...!this.noMinify && {
-            optimization: {
-              minimizer: [
-                new TerserPlugin({
-                  cache: false,
-                  extractComments: false,
-                  terserOptions: {
-                    ecma: 8,
-                  },
-                }) as WebpackPlugin,
-              ],
-            },
-          },
-
-          output: {
-            filename: path.basename(output),
-            path: path.dirname(output),
-          },
-
-          resolve: {
-            alias: {
-              [path.resolve(basedir, `./sources/tools/getPluginConfiguration.ts`)]: path.resolve(basedir, `./sources/tools/getPluginConfiguration.val.js`),
-            },
-          },
-
-          module: {
-            rules: [{
-            // This file is particular in that it exposes the bundle
-            // configuration to the bundle itself (primitive introspection).
-              test: /[\\/]getPluginConfiguration\.val\.js$/,
-              use: {
-                loader: require.resolve(`val-loader`),
-                options: {modules, plugins},
-              },
-            }],
-          },
-
-          plugins: [
-            // esprima is only needed for parsing !!js/function, which isn't part of the FAILSAFE_SCHEMA.
-            // Unfortunately, js-yaml declares it as a hard dependency and requires the entire module,
-            // which causes webpack to add 0.13 MB of unused code to the bundle.
-            // Fortunately, js-yaml wraps the require call inside a try / catch block, so we can just ignore it.
-            // Reference: https://github.com/nodeca/js-yaml/blob/34e5072f43fd36b08aaaad433da73c10d47c41e5/lib/js-yaml/type/js/function.js#L15
-            new webpack.IgnorePlugin({
-              resourceRegExp: /^esprima$/,
-              contextRegExp: /js-yaml/,
-            }),
```

Comment:
This is most likely why the size went pu

Reply:
Esprima isn't in the bundle. I suspect it's because ESBuild uses the browser version of js-yaml rather than the Node one, so Esprima isn't requested. I don't remember what the problem was with the browser version - things seem to work 🤔

Reply:
We shouldn't let it use the browser version of anything as we can pretty much be certain those are polyfilled into oblivion